service/src/main/kotlin/fi/espoo/evaka/vasu/VasuTemplateController.kt

```diff
@@ -44,10 +45,28 @@ class VasuTemplateController {
         }
     }
 
+    @PutMapping("/{id}")
+    fun editTemplate(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable id: UUID,
+        @RequestBody body: VasuTemplateUpdate
+    ) {
+        Audit.VasuTemplateEdit.log()
+        user.requireOneOfRoles(UserRole.ADMIN)
+
+        db.transaction { tx ->
+            val template = tx.getVasuTemplateForUpdate(id) ?: throw NotFound("Template not found")
+            validateTemplateUpdate(template, body)
+            tx.updateVasuTemplate(id, body)
```

Comment:
I'm a bit unsure if editing name/language is fine. Should that be allowed only when document count is zero?

Reply:
Yes, maybe. We should probably also add some kind of a "publish state" enum which would indicate that the template is ready for use. Otherwise we might end up with bogus templates when someone starts using a template that is actually not ready to be used yet.

Reply:
Hmm, well I think in practise the start date is 1.8. and it will be finished before that, so might not be a problem.

Reply:
Name and language are now noneditable if the template is in use.